llama_index/indices/multi_modal/base.py

```diff
@@ -151,12 +151,25 @@ def _get_node_with_embedding(
         Embeddings are called in batches.
 
         """
+        id_to_text_embed_map = None
+
         if is_image:
             id_to_embed_map = embed_image_nodes(
                 nodes,
                 embed_model=self._image_embed_model,
                 show_progress=show_progress,
             )
+
+            # text field is populate, so embed them
+            if all(node.text for node in nodes):
+                id_to_text_embed_map = embed_nodes(
+                    nodes,
+                    embed_model=self._service_context.embed_model,
+                    show_progress=show_progress,
+                )
+                # TODO: refactor this change of image embed model to same as text
+                self._image_embed_model = self._service_context.embed_model
+
```

Comment:
@hatianzhang (CC: @logan-markewich)

this is where I've implemented logic to embed images that have text field populated.

Reply:
lgtm, could we add a flag `is_image_to_text` flag into `_get_node_with_embedding`
if it is true, we embed nodes using text encoder

Reply:
okay if we want to do this, then i should move the logic for inferring if it is `is_image_to_text` outside of `_get_node_with_embedding`